libraries/chain/include/eosio/chain/database_utils.hpp

```diff
@@ -106,35 +106,61 @@ namespace fc {
       from_variant(v, oid._id);
    }
 
+   inline
+   void float64_to_double (const float64_t& f, double& d) {
+      memcpy(&d, &f, sizeof(d));
+   }
+
+   inline
+   void double_to_float64 (const double& d, float64_t& f) {
+      memcpy(&f, &d, sizeof(f));
+   }
+
+   inline
+   void float128_to_uint128 (const float128_t& f, eosio::chain::uint128_t& u) {
+      memcpy(&u, &f, sizeof(u));
+   }
+
+   inline
+   void uint128_to_float128 (const eosio::chain::uint128_t& u,  float128_t& f) {
+      memcpy(&f, &u, sizeof(f));
+   }
+
    inline
    void to_variant( const float64_t& f, variant& v ) {
-      v = variant(*reinterpret_cast<const double*>(&f));
+      double double_f;
+      float64_to_double(f, double_f);
+      v = variant(double_f);
    }
 
    inline
    void from_variant( const variant& v, float64_t& f ) {
-      from_variant(v, *reinterpret_cast<double*>(&f));
+      double double_f;
+      from_variant(v, double_f);
+      double_to_float64(double_f, f);
    }
 
    inline
    void to_variant( const float128_t& f, variant& v ) {
       // Assumes platform is little endian and hex representation of 128-bit integer is in little endian order.	
-      const eosio::chain::uint128_t as_bytes = *reinterpret_cast<const eosio::chain::uint128_t*>(&f);
+      char as_bytes[sizeof(eosio::chain::uint128_t)];
+      memcpy(as_bytes, &f, sizeof(as_bytes));
       std::string s = "0x";	
-      s.append( to_hex( reinterpret_cast<const char*>(&as_bytes), sizeof(as_bytes) ) );	
```

Comment:
Cast to `const char*` is legal and doesn't require memcpy.  This could just drop `as_bytes` and cast `&f`.

Reply:
Are you suggesting to change it? I'm mostly indifferent, as it was already approved previously like this.

Reply:
Up to you.  It's not a big deal, and the code works either way.